Overview.bs

```diff
@@ -623,13 +626,479 @@ The algorithm:
 
 ### Patch Map Table: Format 2 ### {#patch-map-format-2}
 
-<!-- TODO add algorithm explaining how to union the two tables -->
-Coming soon.
+<!-- TODO: clean up unreferenced definitions -->
 
-<!-- 
-TODO:
-- how does this interact with woff2
--->
+<dfn>Format 2 Patch Map</dfn> encoding:
+
+<table>
+  <tr>
+    <th>Type</th><th>Name</th><th>Description</th>
+  </tr>
+  <tr>
+    <td>uint8</td>
+    <td><dfn for="Format 2 Patch Map">format</dfn></td>
+    <td>Set to 2, identifies this as format 2.</td>
+  </tr>
+  <tr>
+    <td>uint32</td>
+    <td><dfn for="Format 2 Patch Map">reserved</dfn></td>
+    <td>Not used, set to 0.</td>
+  </tr>
+  <tr>
+    <td>uint32</td>
+    <td><dfn for="Format 2 Patch Map">id</dfn>[4]</td>
+    <td>Unique ID used to identify patches that are compatible with this font.</td>
+  </tr>
+  <tr>
+    <td>uint8</td>
+    <td><dfn for="Format 2 Patch Map">defaultPatchEncoding</dfn></td>
+    <td>
+      Specifies the format of the patches linked to by uriTemplate (unless overridden by an entry). Uses the ID numbers from the
+      [[#font-patch-formats-summary]] table.
+    </td>
+  </tr>
+  <tr>
+    <td>uint16</td> <!-- TODO: uint32 to match format 1? -->
+    <td><dfn for="Format 2 Patch Map">entryCount</dfn></td>
+    <td>Number of entries encoded in this table.</td>
+  </tr>
+  <tr>
+    <td>Offset32</td>
+    <td><dfn for="Format 2 Patch Map">entries</dfn></td>
+    <td>Offset to a [=Mapping Entries=] sub table. Offset is from the start of this table.</td>
+  </tr>
+  <tr>
+    <td>uint16</td>
+    <td><dfn for="Format 2 Patch Map">uriTemplateLength</dfn></td>
+    <td>
+      Length of the uriTemplate string.
+    </td>
+  </tr>
+  <tr>
+    <td>uint8</td>
+    <td><dfn for="Format 2 Patch Map">uriTemplate</dfn>[uriTemplateLength]</td>
+    <td>
+      A [[!UTF-8]] encoded string. Contains a [[#uri-templates]] which is used to produce URIs associated with each entry.
+    </td>
+  </tr>
+</table>
+
+<dfn>Mapping Entries</dfn> encoding:
```

Comment:
I'm not sure how I feel about these variable-length subtables being packed in this way, e.g. whether it might just make more sense to have offsets. But if we are going to do it this way, I think pretending they're records with this `entries[entryCount]` notation is misleading.  The OpenType spec would do something like "mappingEntries (variable)`.

One aspect of this is that this format seems like it could never be used directly -- you'll always have to unpack it into a separate representation in memory and then use that. Are we sure we're happy with that to save the extra bytes it would take to support direct use? Some of that will be regained by the compression. Is saving the cache space but paying for it in memory for the separate representation a false economy?

Reply:
Makes sense.

Reply:
Changed to be uint8 entries[variable].